syntax = "proto3";

import "common.proto";
import "obstacle_raw.proto";
import "real3d.proto";
import "sync_cylinder_parsing.proto";
import "perception_base.proto";

package autodrive.workflow.proto;

message CameraValidFlag {
  perception_base.ImageId sensor_id = 1;
  bool obstacle_raw_data_valid = 2;
  bool real3d_data_valid = 3;
  bool parsing_data_valid = 4;
}

message CylinderMultiTask {
  // Header
  MsgHeader header = 1;
  // Error
  Error error = 2;
  repeated CameraValidFlag camera_flag_data = 3;
  repeated ObstacleRawsData obstacle_raw_data = 4;
  repeated Real3DData real3d_data = 5;
  repeated CylinderParsingImageData parsing_data = 6;
}

